src/InstrumentationEngine.Attach/Program.cs

```diff
@@ -136,26 +225,91 @@ internal static int Main(string[] args)
                 return ExitCodeFailure;
             }
 
+            #endregion
+
+            #region Profiler Attach
             // CONSIDER: Should the engine be validated before attempting to attach it to the CLR?
             // For example, could check the signature on the assembly (which wouldn't work on Linux).
 
             DiagnosticsClient client = new DiagnosticsClient(processId);
             try
             {
-                client.AttachProfiler(TimeSpan.FromSeconds(10), InstrumentationEngineClsid, enginePath);
+                client.AttachProfiler(TimeSpan.FromSeconds(10), InstrumentationEngineClsid, enginePath, bytes);
             }
             catch (ServerErrorException ex)
             {
                 WriteError(Invariant($"Could not attach engine to process: '{ex.Message}'."));
                 return ExitCodeFailure;
             }
 
+            #endregion
+
             return ExitCodeSuccess;
         }
 
         private static void WriteError(string message)
         {
             Console.Error.WriteLine(message);
         }
+
+        private static bool TryParseEngineConfiguration(
+            InstrumentationConfigurationSources sources,
+            Chip targetChip,
+            out InstrumentationEngineConfiguration configuration)
+        {
+            var engineSection = new InstrumentationEngineConfigurationInstrumentationEngine();
+            var methodsSection = new List<instrumentationMethodsTypeAddInstrumentationMethod>();
+
+            // Parse engine settings
+            var engineSettings = new List<settingsTypeSetting>()
+            {
+                new settingsTypeSetting()
+                {
+                    Name = "LogLevel",
+                    Value = "Errors"
+                }
+            };
+            engineSection.Settings = engineSettings.ToArray();
+
+            // Parse methods section
+            foreach (var source in sources.InstrumentationConfigurationSource)
+            {
+                foreach (var platform in source.Platforms)
+                {
+                    if (platform.Chip.Equals(targetChip.ToString("G"), StringComparison.OrdinalIgnoreCase))
+                    {
+                        var methodSettings = new List<settingsTypeSetting>();
+                        var methodSection = new instrumentationMethodsTypeAddInstrumentationMethod()
+                        {
+                            ConfigPath = platform.Path
+                        };
+
+                        foreach (var setting in source.Settings)
+                        {
+                            methodSettings.Add(new settingsTypeSetting()
+                            {
+                                Name = setting.Name,
+                                Value = setting.Value
```

Comment:
>Value [](start = 48, length = 5)

We might need to some type of transform on these values to do token replacement, but let's leave that for a different change once we figure out the token mechanism and the valid token values.